backend/app/api/endpoints/projects.py

```
@@ -0,0 +1,241 @@
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail="Project not found",
+        )
+
+    return project
+
+
+@router.put("/{project_id}", response_model=ProjectResponse)
+def update_project_endpoint(
+    project_id: int = Path(..., description="Project ID"),
+    project_update: ProjectUpdate = None,
+    current_user: User = Depends(get_current_user),
```

Ruff (0.14.10)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)